test/stableTRIStaking/migrate.test.ts

```diff
@@ -0,0 +1,80 @@
+import { ethers } from "hardhat";
+import chai from "chai";
+import { solidity } from "ethereum-waffle";
+import { Tri__factory } from "../../typechain";
+import { getBigNumber } from "../utils";
+
+chai.use(solidity);
+const { expect } = chai;
+
+describe("RevenueDistributionToken - Migrate", function () {
+  beforeEach(async function () {
+    const [deployer, user0] = await ethers.getSigners();
+    this.deployer = deployer;
+    this.user0 = user0;
+    const triFactory = new Tri__factory(this.deployer);
+    this.tri = await triFactory.deploy(this.deployer.address);
+    await this.tri.mint(this.deployer.address, "100");
+    this.xTRI = await ethers.getContractFactory("TriBar");
+    this.xTRI = await this.xTRI.deploy(this.tri.address);
+    await this.tri.approve(this.xTRI.address, "100");
+    await this.xTRI.enter("100");
+    this.pTRI = await ethers.getContractFactory("StableTRIStaking");
+    const ERC20Mock = await ethers.getContractFactory("ERC20Mock");
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    this.rewardToken = await ERC20Mock.deploy("rewardToken", "rewardToken", 18, getBigNumber("300") as any);
+    const pTRIConstructorArgs = [
+      "pTRI",
+      "pTRI",
+      this.rewardToken.address,
+      this.tri.address,
+      this.deployer.address,
+      0,
+    ];
+    this.pTRI = await this.pTRI.deploy(...pTRIConstructorArgs);
+    await this.xTRI.approve(this.pTRI.address, "100");
+  });
+
+  it("should migrate", async function () {
```

Comment:
can you add a test when user has already deposited in the pTRI vault
some time has passed and then the user migrates

check for values of rewardTokens received by the user after migration